[nnc] Updated exceptions to add a generic error message to turn off fuser (1/n)

navahgar opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #64329
* #64328
* #64319
* #64318
* __->__ #64317
* #64316

Differential Revision: [D30683944](https://our.internmc.facebook.com/intern/diff/D30683944)

navahgar committed:
[nnc] Updated exceptions to add a generic error message to turn off fuser (1/n)

[ghstack-poisoned]

navahgar commented:
@navahgar has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30683944).

bertmaher commented on torch/csrc/jit/tensorexpr/block_codegen.cpp:
```diff
@@ -49,7 +49,8 @@ BufPtr BlockAnalysis::getMultiDimBuf(BufPtr buf) const {
     return input_->second;
   } else {
-    throw std::runtime_error("BlockCodeGen: Entry not in input/Buffer map");
+    throw std::runtime_error(
```
Can we turn these into TORCH_INTERNAL_ASSERT?

bertmaher commented on torch/csrc/jit/tensorexpr/cuda_codegen.cpp:
```diff
@@ -303,7 +309,7 @@ void CudaPrinter::visit(IntrinsicsPtr v) {
 
 void CudaPrinter::visit(ExternalCallPtr v) {
-  throw unimplemented_lowering(v);
+  throw unimplemented_lowering(buildErrorMessage(std::to_string(v)));
```
Why not have the `unimplemented_lowering` constructor call `buildErrorMessage` instead of doing it at every call site?

navahgar committed:
Update on "[nnc] Updated exceptions to add a generic error message to turn off fuser (1/n)"


Differential Revision: [D30683944](https://our.internmc.facebook.com/intern/diff/D30683944)

[ghstack-poisoned]

navahgar commented:
@navahgar has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30683944).
